### PR TITLE
Integrate HTTP adapter with Reconciler

### PR DESCRIPTION
**Pull request by antoineco**
**Description**

Integrate HTTP adapter with Reconciler. Round 2 of #6236

Changes proposed in this pull request:

* Wait for Channel readiness before synchronizing adapter Service
* Add Ports and EnvVars to adapter Service
* Pass metrics and logging configs to adapter Service
* Configure ReadinessProbe of adapter Service
* Inject adapter image via envconfig instead of os.Getenv()
* Refactor some of the reconciliation flow

**Related issue(s)**

#6236
#6175

### CODE REVIEW

File: components/event-sources/reconciler/httpsource/adapter_config.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"knative.dev/pkg/metrics"
+)
+
+const component = "httpsource"
+
+// httpAdapterEnvConfig contains properties used to configure the HTTP receive
+// adapter. These are automatically populated by envconfig.
+type httpAdapterEnvConfig struct {
+	// Container image
+	Image string `required:"true"`
```
[the remark this answers is not in the document]
Comment (antoineco) [reply components/event-sources/reconciler/httpsource/adapter_config.go]: Done

File: components/event-sources/config/500-sources-controller.yaml [hunk 2]
```diff
@@ -48,4 +48,4 @@ spec:
           - name: METRICS_DOMAIN
             value: kyma-project.io/event-sources
           - name: HTTP_ADAPTER_IMAGE
-            value: github.com/kyma-project/kyma/components/event-sources/cmd/httpsource-adapter
+            value: github.com/kyma-project/kyma/components/event-sources/cmd/http-adapter
```
Comment (k15r) [inline components/event-sources/config/500-sources-controller.yaml]: to make it clearer that these parts will be replaces by ko we should use strict references
means: `ko://github.com/kyma-project/kyma/components/event-sources/cmd/http-adapter`.

Comment (antoineco) [reply components/event-sources/config/500-sources-controller.yaml]: Didn't know that trick, thanks! (and done)

File: components/event-sources/reconciler/objects/service.go [hunk 3]
```diff
@@ -54,66 +54,84 @@ type ServiceOption func(*servingv1alpha1.Service)
...
-		}
-		(*containers)[0].Image = img
+// WithContainerPort sets the container port of a Service.
+func WithContainerPort(port int32) ServiceOption {
+	return func(s *servingv1alpha1.Service) {
+		ports := &firstServiceContainer(s).Ports
+
+		*ports = append(*ports, corev1.ContainerPort{
+			// empty name defaults to http/1.1 protocol
+			ContainerPort: port,
```
Comment (k15r) [inline components/event-sources/reconciler/objects/service.go]: please add a name for the port. Unnamed port cause weird issues with istio

Comment (antoineco) [reply components/event-sources/reconciler/objects/service.go]: It's illegal in a Kn Service, unless you name it exactly "h2c" to force http2.

Comment (k15r) [reply components/event-sources/reconciler/objects/service.go]: oki doki

Comment (antoineco) [reply components/event-sources/reconciler/objects/service.go]: No worries, Kn _does_ set a name on the Pod itself. This constructor only sets the port on the Kn Service, which gets converted to a deployment with all the missing stuff injected.

**Pull request merged**